src/utils_GUI/results_axis_utils.jl

```diff
@@ -219,14 +219,14 @@ function get_periods(T::TS.TimeStructure, type::Type, sp::Int64, rp::Int64, sc::
             if eltype(T.operational[sp].rep_periods) <: TS.OperationalScenarios
                 return [
                     t for
-                    t ∈ T if t.sp == sp && t.period.rp == rp && t.period.period.sc == sc
+                    t ∈ T if t.sp == sp && t.period.rp == rp && t.period.period.osc == sc
```

Comment:
We have to try to find a different design here. Although I am quite certain we are in `TimeStruct` now at a state that we do no longer change it, it is still problematic to go directly for the individual fields. This is relevant for all methods for the function `get_periods()`.

On a side note, do we still need `get_periods(T::TS.TimeStructure, type::Type)`. It is at least not called within `results_axis_utils.jl`

Reply:
Yes, I considered using internal functions in TimeStruct for this, but I guess that is also problematic?

`get_periods(T::TS.TimeStructure, type::Type)` is called in `src/utils_GUI/GUI_utils.jl`.

Reply:
Yeah, it is equally problematic. There are other approaches how we can handle that using the Parametric input type. We can revisit it later.